Sources/Networking/HTTPClient.swift

```diff
@@ -18,23 +18,29 @@ class HTTPClient {
     typealias RequestHeaders = [String: String]
     typealias Completion<Value: HTTPResponseBody> = (HTTPResponse<Value>.Result) -> Void
 
-    private let session: URLSession
     internal let systemInfo: SystemInfo
+    internal let timeout: TimeInterval
+
+    private let session: URLSession
     private let state: Atomic<State> = .init(.initial)
     private let eTagManager: ETagManager
     private let dnsChecker: DNSCheckerType.Type
 
     init(
         systemInfo: SystemInfo,
         eTagManager: ETagManager,
-        dnsChecker: DNSCheckerType.Type = DNSChecker.self
+        dnsChecker: DNSCheckerType.Type = DNSChecker.self,
+        timeoutSeconds: TimeInterval = Configuration.networkTimeoutDefault
     ) {
         let config = URLSessionConfiguration.ephemeral
         config.httpMaximumConnectionsPerHost = 1
+        config.timeoutIntervalForRequest = timeoutSeconds
+        config.timeoutIntervalForResource = timeoutSeconds
```

Comment:
👍🏻 I can't believe we didn't have these.

Reply:
Hah, same